app/vlinsert/opentelemetry/pb.go

```diff
@@ -0,0 +1,285 @@
+		src, err = fc.NextField(src)
+		if err != nil {
+			return fmt.Errorf("cannot read next field in ExportLogsServiceRequest: %w", err)
+		}
+		switch fc.FieldNum {
+		case 1:
+			data, ok := fc.MessageData()
+			if !ok {
+				return fmt.Errorf("cannot read ResourceLogs data")
+			}
+			r.ResourceLogs = append(r.ResourceLogs, &ResourceLogs{})
+			rm := r.ResourceLogs[len(r.ResourceLogs)-1]
+			if err := rm.unmarshalProtobuf(data); err != nil {
+				return fmt.Errorf("cannot unmarshal ResourceLogs: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+// ResourceLogs represents the corresponding OTEL protobuf message
+type ResourceLogs struct {
+	Resource  *pb.Resource
+	ScopeLogs []*ScopeLogs
+}
+
+func (rl *ResourceLogs) unmarshalProtobuf(src []byte) (err error) {
+	// message ResourceLogs {
+	//   Resource resource = 1;
+	//   repeated ScopeLogs scope_logs = 2;
+	// }
+	var fc easyproto.FieldContext
+	for len(src) > 0 {
+		src, err = fc.NextField(src)
+		if err != nil {
+			return fmt.Errorf("cannot read next field in ResourceLogs: %w", err)
+		}
+		switch fc.FieldNum {
+		case 1:
+			data, ok := fc.MessageData()
+			if !ok {
+				return fmt.Errorf("cannot read Resource data")
+			}
+			rl.Resource = &pb.Resource{}
+			if err := rl.Resource.UnmarshalProtobuf(data); err != nil {
+				return fmt.Errorf("cannot umarshal Resource: %w", err)
+			}
+		case 2:
+			data, ok := fc.MessageData()
+			if !ok {
+				return fmt.Errorf("cannot read ScopeLogs data")
+			}
+			rl.ScopeLogs = append(rl.ScopeLogs, &ScopeLogs{})
+			sl := rl.ScopeLogs[len(rl.ScopeLogs)-1]
+			if err := sl.unmarshalProtobuf(data); err != nil {
+				return fmt.Errorf("cannot unmarshal ScopeLogs: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+// ScopeLogs represents the corresponding OTEL protobuf message
+type ScopeLogs struct {
+	LogRecords []*LogRecord
+}
+
+func (sl *ScopeLogs) unmarshalProtobuf(src []byte) (err error) {
+	// message ScopeLogs {
+	//   repeated LogRecord log_records = 2;
+	// }
+	var fc easyproto.FieldContext
+	for len(src) > 0 {
+		src, err = fc.NextField(src)
+		if err != nil {
+			return fmt.Errorf("cannot read next field in ScopeLogs: %w", err)
+		}
+		switch fc.FieldNum {
+		case 2:
+			data, ok := fc.MessageData()
+			if !ok {
+				return fmt.Errorf("cannot read LogRecord data")
+			}
+			sl.LogRecords = append(sl.LogRecords, &LogRecord{})
+			l := sl.LogRecords[len(sl.LogRecords)-1]
+			if err := l.unmarshalProtobuf(data); err != nil {
+				return fmt.Errorf("cannot unmarshal LogRecord: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+// LogRecord represents the corresponding OTEL protobuf message
+type LogRecord struct {
+	TimeUnixNano         pb.Uint64
+	ObservedTimeUnixNano pb.Uint64
+	SeverityNumber       int32
+	SeverityText         string
+	Body                 *pb.AnyValue
+	Attributes           []*pb.KeyValue
+}
+
+func (r *LogRecord) severity() string {
+	if r.SeverityText != "" {
+		return r.SeverityText
+	}
+	return logSeverities[r.SeverityNumber]
```

Comment:
Could it panic?

Reply:
it should not, it covers all subset of allowed values defined in spec, but i can also account it